homeassistant/components/netatmo/climate.py

```diff
@@ -270,35 +280,37 @@ async def handle_event(self, event):
                 self._target_temperature = self._away_temperature
             elif self._preset == PRESET_SCHEDULE:
                 self.async_update_callback()
+                self.data_handler.async_force_update(self._home_status_class)
             self.async_write_ha_state()
             return
 
-        if not home.get("rooms"):
-            return
-
-        for room in home["rooms"]:
-            if data["event_type"] == EVENT_TYPE_SET_POINT:
-                if self._id == room["id"]:
-                    if room["therm_setpoint_mode"] == STATE_NETATMO_OFF:
-                        self._hvac_mode = HVAC_MODE_OFF
-                    elif room["therm_setpoint_mode"] == STATE_NETATMO_MAX:
-                        self._hvac_mode = HVAC_MODE_HEAT
-                        self._target_temperature = DEFAULT_MAX_TEMP
-                    elif room["therm_setpoint_mode"] == STATE_NETATMO_MANUAL:
+        for room in home.get("rooms"):
```

Comment:
`get()` may return None, which is not iteratable. 
```suggestion
        for room in home.get("rooms", []):
```

Reply:
This shouldn't happen but you're totally right. This does definitely not hurt.